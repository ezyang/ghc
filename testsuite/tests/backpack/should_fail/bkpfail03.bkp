unit q where
    module M1 where
        data M = M
    signature M2(M) where
        import M1
unit p where
    module M2 where
        data M = M
    include q
