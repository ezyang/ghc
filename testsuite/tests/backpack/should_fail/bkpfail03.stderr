[1 of 2] Processing q
  [1 of 3] Compiling M1               ( q/M1.hs, nothing )
  [2 of 3] Compiling M2[boot]         ( q/M2.hs-boot, nothing )
  [3 of 3] Compiling M2[merge]        ( q/M2.hs-boot-merge, nothing )
[2 of 2] Processing p

bkpfail03.bkp:9:5: error:
    Failed to unify when filling requirements:
    Context:
      provides: M2 -> p():M2
                      M{M, M}
      requires:
    Merging shape:
      provides: M1 -> q(hole:M2):M1
                      M{M, M}
      requires: M2 -> q(hole:M2):M1.M{M}
